javascript/ecmascript-2015/intro-ecmascript/const-part-1.md

```diff
@@ -45,18 +45,35 @@ const year = 2016;
 Once a `const` is declared any attempts to reassign the value will cause an error:
 
 ```javascript
-year = 2017; //error
-year = "future"; //error
+year = 2017; // error
+year = "future"; // error
 ```
 
-`const`s can exist in different scopes for example if we had already declared a `const` variable called `year` it is valid to declare another as long as it exists in another scope:
+Before we move on with this insight, we need to take a look at what scope means in JavaScript. Scope is the accessibility of variables, functions, and objects in some particular part of your code during runtime. In other words, scope determines the visibility of variables and other resources in areas of your code. Here is a simple example that should clear any confusion:
+
+```js
+function foo() {
+  var x = "declared inside function";
+  // x can only be used in foo
+  console.log(x);
+}
+
+console.log(x);
+// Throws an error
+```
+
+Now that you understand what a scope is, it is important to know that `const`s can exist in different scopes. For example, if we had already declared a `const` variable called `year`, it is valid to declare another as long as it exists in another scope:
```

Comment:
I think the flip-side to this explanation might make more sense? What do you think? 

```suggestion
Now that you understand what scope is, it is important to know that `const`s cannot exist in the same scope. For example, if we had already declared a `const` variable called `year`, we cannot declare another one in the same scope, but we are free to do so in any other scope:
```

With the above change we would also show the error case in the code example:

```javascript
const year = 2015;
console.log(year);
// 2015

// ...

// declaring a const variable in the same 
// scope would cause a syntax error
// "Identifier 'year' has already been declared"
const year = 2016;

if (1 == 1) {
  // However, it is perfectly fine to do
  // so in a different scope
  const year = 2016;
  console.log(year);
  // 2016
}

Reply:
yeah, this looks much better